lectures/symbolic_sympy.md

```diff
@@ -0,0 +1,272 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.5
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Introduction to Symbolic Algebra and Calculus with Sympy
+
+
+## Overview
+
+Symbolic computation, or computer algebra, involves using algorithms and software to perform exact computations and manipulate mathematical equations. While **Mathematica** is a popular tool for symbolic computations, it's proprietary and can be costly. As an open-source alternative, the **sympy** library in Python offers a comprehensive set of functionalities for symbolic mathematics. This allows users to perform a range of operations, from algebraic manipulations to calculus, providing exact solutions instead of numerical approximations.
+
+
+## Getting Started
+
+Let's get started with **sympy** by installing and setting up the package.
+
+```{code-cell} ipython3
+! pip install sympy
+```
+
+We'll start by importing the necessary **sympy** functions and initializing our environment.
+
+```{code-cell} ipython3
+from sympy import symbols, init_printing
+
+# Enable best printer available
+init_printing()
+
+# Let's define some symbols to work with
+x, y, z = symbols('x y z')
+```
+
+We can now use these symbols **x**, **y**, and **z** to build symbolic expressions.
+
+```{code-cell} ipython3
+---
+mystnb:
+  image:
+    width: 10%
+---
+# An expression
+expr = (x + y) ** 2
+expr
+```
+
+## Symbolic Algebra
+
+
+### Algebraic Expressions
+
+**sympy** provides several functions to create and manipulate algebraic expressions. Let's look at a few examples.
+
+Creating Algebraic Expressions
```

Comment:
Please use sentences when describing the code (i.e., 

"Here we create a quadratic equation with variable `x`")

If a code block directly follows the sentence, no full stop is needed.